x-pack/plugins/enterprise_search/public/applications/app_search/components/search_ui/search_ui_logic.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { flashAPIErrors } from '../../../shared/flash_messages';
+import { HttpLogic } from '../../../shared/http';
+import { EngineLogic } from '../engine';
+
+interface InitialFieldValues {
+  validFields: string[];
+  validSortFields: string[];
+  validFacetFields: string[];
+}
+interface SearchUIActions {
+  initializeData(): void;
+  dataInitialized(initialFieldValues: InitialFieldValues): InitialFieldValues;
+  titleFieldChanged(titleField: string): { titleField: string };
+  URLFieldChanged(urlField: string): { urlField: string };
+  facetFieldsChanged(facetFields: string[]): { facetFields: string[] };
+  sortFieldsChanged(sortFields: string[]): { sortFields: string[] };
+  activeFieldChanged(activeField: string): { activeField: string };
+}
+
+interface SearchUIValues {
+  dataLoading: boolean;
+  validFields: string[];
+  validSortFields: string[];
+  validFacetFields: string[];
+  titleField: string;
+  urlField: string;
+  facetFields: string[];
+  sortFields: string[];
+  activeField: string;
```

Comment:
[optional] might be helpful to make this a string enum? I'm guessing this is the logic used for the custom SVG highlighting, right?

```suggestion
  activeField: 'title' | 'url' | 'facet' | ' sort';
```

Reply:
5df56d451eec668fd7baf40ca3ef921bda132ba3

Reply:
Haha sorry for any enum annoyingness here, totally not my intention 🤦‍♀️